### PR TITLE
Implemented task cancellation.

### LINKED ISSUE #36: Implement transfer task cancellation
We should provide an endpoint for canceling a previously created transfer task. Obviously a cancellation shouldn't affect any transfers that are already finished. The state in which the files are left depends on the endpoint implementation, which for the DTS prototype is whatever Globus does when you cancel a transfer task.

We'll want this in place before we turn over the DTS prototype to our stakeholders for testing.

### PR DESCRIPTION
**Pull request by jeff-cohere**
I also fixed some glitches in the Globus transfer tests.

Closes #36

### CODE REVIEW

**Comment by jeff-cohere** [comment]
Okay, this PR's ready to go.

**Comment by jeff-cohere** [comment]
I've identified a case in which a task may be cancelled and its status may not be properly updated afterward, so I'm working on a fix. Should only affect one file, though.

**Comment by jeff-cohere** [comment]
Okay, sorry about the wobbles. Let's try this out. I'll turn my attention next to some code cleanup.

File: endpoints/globus/endpoint.go [hunk 1]
```diff
@@ -317,51 +318,35 @@ func (ep *Endpoint) Transfers() ([]uuid.UUID, error) {
...
-	if !staged {
-		return uuid.UUID{}, fmt.Errorf("The files requested for transfer are not yet staged.")
-	}
-
-	// obtain a submission ID
-	// https://docs.globus.org/api/transfer/task_submit/#get_submission_id
-	var xferId uuid.UUID
+// https://docs.globus.org/api/transfer/task_submit/#get_submission_id
+func (ep *Endpoint) getSubmissionId() (uuid.UUID, error) {
+	var id uuid.UUID
```
Comment (ialarmedalien) [inline endpoints/globus/endpoint.go:323]: any reason for the variable name change? `xferId` would be consistent with the next function

Comment (jeff-cohere) [reply endpoints/globus/endpoint.go:323]: Sorry for the confusion--these functions are different. `getSubmissionId()` returns an ID for submitting a Globus transfer task, after which a transfer task ID is issued.

**Review by ialarmedalien — APPROVED** [review]

File: endpoints/globus/endpoint.go (hunk 1 above)
Comment (ialarmedalien) [reply endpoints/globus/endpoint.go:323]: Ah, got it. Wasn't sure if submission ID == transfer ID or not.

**Pull request merged**